R/colocalization_heatmap.R

```diff
@@ -45,13 +155,22 @@ ColocalizationHeatmap <- function(
   if (!inherits(data, what = "tbl_df")) {
     abort(glue("'data' must be a 'tbl_df' object"))
   }
+  type <- match.arg(type, choices = c("tiles", "dots"))
   stopifnot(
     "'marker1_col' is not in 'data'" =
       marker1_col %in% names(data),
     "'marker2_col' is not in 'data'" =
       marker2_col %in% names(data),
     "'value_col' is not in 'data'" =
       value_col %in% names(data),
+    "'size_col' is not in data" =
+      size_col %in% names(data),
+    "'size_range' must be a numeric vector with positive values of length 2" =
+      (inherits(size_range, what = "numeric") &&
+        length(size_range) == 2) &&
+        all(size_range > 0),
```

Comment:
Could it be useful to allow to set 0 as well? It is valid as input in ggplot
`all(size_range >= 0)`

Reply:
Yeah I wasn't sure because maybe it's a little surprising if some dots disappear completely but I'll change it!